### PR TITLE
Updated Contributing Guidelines

### PR DESCRIPTION
**Pull request by HarryWangATX**

### CODE REVIEW

File: content/1_General/Contributing.mdx [hunk 1]
```diff
@@ -87,6 +87,18 @@ Some examples:
...
+## Code Conventions
+
+Please follow these code conventions when contributing a solution to USACO Guide:
+
+- Try to use descriptive variable names.
+- Use `snake_case` for Python.
+- Do not name a data structure the data structure's name.
+- If you use a I/O template, make sure to collapse it with `CodeSnips`.
+  - Start your `CodeSnip` with `//BeginCodeSnip{I/O Template}`.
+  - End your `CodeSnip` with `//EndCodeSnip`.
```
Comment (dongliu0426) [inline content/1_General/Contributing.mdx:99]: just link to the codesnip docs?

File: content/1_General/Contributing.mdx [hunk 2]
```diff
@@ -87,6 +87,16 @@ Some examples:
   - If no editorial exists, or if existing editorial could be improved.
   - Or solution code in a different language, etc.
 
+## Code Conventions
+
+Please follow these code conventions when contributing a solution to USACO Guide:
+
+- Try to use descriptive variable names.
+- Use `snake_case` for Python.
+- Do not name a data structure the data structure's name.
```
Comment (dongliu0426) [inline content/1_General/Contributing.mdx:96]: example? smth like "e.g. don't name `std::set`s set"

File: content/1_General/Contributing.mdx [hunk 3]
```diff
@@ -87,6 +87,16 @@ Some examples:
...
+- If you use a I/O template, make sure to collapse it with [`CodeSnips`](https://usaco.guide/general/working-mdx?#code-blocks-and-code-snippets).
```
Comment (maggie-j-liu) [inline content/1_General/Contributing.mdx:97]:
```suggestion
- If you use a I/O template, make sure to collapse it with [`CodeSnips`](https://usaco.guide/general/working-mdx#code-blocks-and-code-snippets).
```

**Comment by HarryWangATX** [comment]
i should switch to adding soution file probably

File: content/1_General/Adding_Solution.mdx [hunk 4]
```diff
@@ -178,11 +177,12 @@ have to follow all of the recommendations):
...
-<IncompleteSection>
+### General Conventions
 
-additional resources about style?
-
-</IncompleteSection>
+- Try to use descriptive variable names.
+- Do not name a data structure the data structure's name.
+  - For example, don't name `std::set` set
+- If you use a I/O template, make sure to collapse it with [`CodeSnip`s](https://usaco.guide/general/working-mdx?#code-blocks-and-code-snippets).
```
Comment (maggie-j-liu) [inline content/1_General/Adding_Solution.mdx:185]:
```suggestion
- If you use a I/O template, make sure to collapse it with [`CodeSnip`s](https://usaco.guide/general/working-mdx#code-blocks-and-code-snippets).
```

File: content/1_General/Adding_Solution.mdx [hunk 5]
```diff
@@ -234,15 +234,16 @@ ones listed above. In particular, for loop macros are highly discouraged.
...
+`BufferedReader` and should be avoided. You can also use your own I/O templates, but please follow the general guidelines above regarding I/O templates.
 
 </JavaSection>
 
-<PySection />
+<PySection>
+
+### Python
+
+Please use `snake_case` in your code.
```
Comment (maggie-j-liu) [inline content/1_General/Adding_Solution.mdx:245]:
```suggestion
Please use `snake_case` to name variables in your code.
```

**Review by maggie-j-liu — APPROVED** [review]

File: content/1_General/Adding_Solution.mdx [hunk 6]
```diff
@@ -178,11 +177,12 @@ have to follow all of the recommendations):
...
 
-<IncompleteSection>
+### General Conventions
 
-additional resources about style?
-
-</IncompleteSection>
+- Try to use descriptive variable names.
+- Do not name a data structure the data structure's name.
+  - For example, don't name `std::set` set
```
Comment (dongliu0426) [inline content/1_General/Adding_Solution.mdx:184]:
```suggestion
  - For example, don't name `std::set` "set".
```

**Review by dongliu0426 — APPROVED** [review]

File: content/1_General/Adding_Solution.mdx [hunk 7]
```diff
@@ -178,11 +177,12 @@ have to follow all of the recommendations):
...
 
-<IncompleteSection>
+### General Conventions
 
-additional resources about style?
-
-</IncompleteSection>
+- Try to use descriptive variable names.
+- Do not name a data structure the data structure's name.
+  - For example, don't name `std::set` "set".
```
Comment (jessechoe10) [inline content/1_General/Adding_Solution.mdx:184]: Why this?

Comment (HarryWangATX) [reply content/1_General/Adding_Solution.mdx:184]: We voted in our meeting to adhere to these guidelines for the health of the guide.

**Review by bqi343 — APPROVED** [review]

**Comment by HarryWangATX** [comment]
good to merge after checks?

**Comment by bqi343** [comment]
@thecodingwizard ?

File: content/1_General/Adding_Solution.mdx [hunk 8]
```diff
@@ -150,22 +131,50 @@ See [Working With MDX](/general/working-mdx) for additional information.
...
+ a,b = (0,1)
+ print(a+b)
+ print(a + b)
+```
+
+- Use universally understandable variable names, especially for Bronze and
+  Silver. Should be more descriptive than what you would normally use
+  in-contest.
+  - Do not name a data structure the data structure's name. For example, don't
+    name a `std::set` "set."
```
Comment (thecodingwizard) [inline content/1_General/Adding_Solution.mdx:167]: Shouldn't this be "For C++, do not use names that could conflict with `using namespace std`. Therefore don't name `std::set` a set" or something like that? (At least, that was my intent when proposing this rule)

Comment (HarryWangATX) [reply content/1_General/Adding_Solution.mdx:167]: I thought the intent was just to have names not be the structure's name in any language??

**Pull request merged**